middleware.go

```diff
@@ -4,22 +4,19 @@ import (
 	"net/http"
 )
 
-// NextMiddlewareFunc is the function signature for the next parameter in ServerHTTPMiddleware
-type NextMiddlewareFunc func(http.ResponseWriter, *http.Request)
```

Comment:
I think providing this type definition is convenient. Users of the library still may use it or not – there is actually no cast necessary (see https://golang.org/ref/spec#Type_identity, named vs unnamed types). I wrote a small example: https://play.golang.org/p/2linqiss6S